Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift

```diff
@@ -240,20 +240,34 @@ extension XMLKeyedDecodingContainer {
         }
 
         let elements = container
-            .withShared { keyedBox -> [KeyedBox.Element] in
-                keyedBox.elements[key.stringValue].map {
-                    if let singleKeyed = $0 as? SingleKeyedBox {
-                        return singleKeyed.element.isNull ? singleKeyed : singleKeyed.element
-                    } else {
-                        return $0
+            .withShared { keyedBox -> [KeyedBox.Element] in                
+                if type is XMLPositionIndexedSequenceProtocol.Type {
+                    return keyedBox.elements
+                        .indexedValues(for: key.stringValue)
+                        .map { indexedValue -> Box in
+                            KeyedBox(
+                                elements: [
+                                    ("index", StringBox(xmlString: String(indexedValue.index))),
+                                    ("value", indexedValue.value)
+                                ],
+                                attributes: []
+                            )
+                        }
+                } else {
+                    return keyedBox.elements[key.stringValue].map {
+                        if let singleKeyed = $0 as? SingleKeyedBox {
+                            return singleKeyed.element.isNull ? singleKeyed : singleKeyed.element
+                        } else {
+                            return $0
+                        }
                     }
                 }
             }
-
+        
```

Comment:
Please run SwiftFormat on the source code to clean up trailing whitespaces. You can also install pre-commit hooks for it [as described in the "Coding Style" section](https://github.com/MaxDesiatov/XMLCoder#coding-style) of `README.md`, which will make `swiftformat` run before you commit.